src/components/Leaderboard/DailyLeaderboard.tsx

```diff
@@ -0,0 +1,141 @@
+import { useEffect, useState } from 'react';
+import ReactPaginate from 'react-paginate';
+import { useAppSelector } from '../../store/hooks';
+import styles from './Leaderboard.module.css';
+import {
+  DailyChallengesApi,
+  DailyChallengeLeaderBoardResponse,
+} from '@codecharacter-2023/client';
+import { apiConfig, ApiError } from '../../api/ApiConfig';
+import Loader from '../Loader/Loader';
+import trophyImage from '../../assets/trophy.png';
+import Toast from 'react-hot-toast';
+import { user } from '../../store/User/UserSlice';
+import { Table } from 'react-bootstrap';
+
+function PaginatedItems() {
+  const [pageCount, setPageCount] = useState(0);
+  const [itemOffset, setItemOffset] = useState(0);
+  const [items, setItems] = useState<DailyChallengeLeaderBoardResponse[]>([]);
+  const [currentItems, setCurrentItems] = useState<
+    DailyChallengeLeaderBoardResponse[]
+  >([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const itemsPerPage = 10;
+  const currentUserName = useAppSelector(user).username;
+
+  useEffect(() => {
+    fetchLeaderboard();
+  }, []);
+
+  useEffect(() => {
+    const endOffset = itemOffset + itemsPerPage;
+    setCurrentItems(items.slice(itemOffset, endOffset));
+    setPageCount(Math.ceil(items.length / itemsPerPage));
+  }, [itemOffset, itemsPerPage, items]);
+
+  const handlePageClick = (event: { selected: number }) => {
+    const newOffset = (event.selected * itemsPerPage) % items.length;
+    setItemOffset(newOffset);
+  };
+
+  const fetchLeaderboard = () => {
+    setIsLoaded(false);
+    const leaderboardAPI = new DailyChallengesApi(apiConfig);
+    leaderboardAPI
+      .getDailyChallengeLeaderBoard() // The pagination system is so messed up, I can't be bother to fix it, so this hack
+      .then(response => {
+        setItems(response);
+        setIsLoaded(true);
+      })
+      .catch(error => {
+        if (error instanceof ApiError) Toast.error(error.message);
+      });
+  };
```

Comment:
You can add these redux. no need of storing data. just store status of leaderboard like `Loading`,`Success` and `Error`